homeassistant/components/freedns.py

```diff
@@ -0,0 +1,115 @@
+"""
+Integrate with FreeDNS Dynamic DNS service at freedns.afraid.org.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/freedns/
+"""
+import asyncio
+from datetime import timedelta
+import logging
+
+import aiohttp
+from aiohttp.hdrs import USER_AGENT
+import async_timeout
+import voluptuous as vol
+
+from homeassistant.const import (CONF_URL, CONF_TIMEOUT, CONF_ACCESS_TOKEN)
+from homeassistant.helpers.aiohttp_client import SERVER_SOFTWARE
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'freedns'
+
+# We should set a dedicated address for the user agent.
+EMAIL = 'hello@home-assistant.io'
+
+DEFAULT_INTERVAL = timedelta(minutes=10)
+DEFAULT_TIMEOUT = 10
+
+UPDATE_URL = 'https://freedns.afraid.org/dynamic/update.php'
+HA_USER_AGENT = "{} {}".format(SERVER_SOFTWARE, EMAIL)
```

Comment:
Do we need this user agent header? We set already a default homeassistant agent

Reply:
Depends. At least NO-IP requires that there is an e-mail address attached to the request.